src/main/java/org/alfresco/service/namespace/QName.java

```diff
@@ -215,7 +219,28 @@ public static String createValidLocalName(String name)
 
         return name;
     }
-    
+
+    /**
+     * Check for special characters in name
+     *
+     * @param name  name to create valid local name from
+     * @throws IllegalArgumentException
+     */
+    public static void checkName(String name)
```

Comment:
I would like to propose to do some extra for this method.
Either rename it to something like "checkLocalNameForInvalidCharacters" as there are other checks in the code like the check for null in "createQName" methods or move the other checks  ("(localName == null || localName.length() == 0)) into this method and rename it to "checkLocalName"
I would prefer the second.

The other note is that if you have an exception thrown in this method, the signature should mention that. You may have noticed that the checks for non present local name throw InvalidQNameException whereas your method throw a different IllegalArgumentException. I would make this exception the same.

Reply:
Sure.  Thanks for the great pointers.